cockatrice/src/client/ui/widgets/cards/card_info_picture_widget.cpp

```diff
@@ -0,0 +1,219 @@
+}
+
+/**
+ * @brief Marks the pixmap as dirty and triggers a widget repaint.
+ *
+ * Sets `pixmapDirty` to true, indicating that the pixmap needs to be reloaded before the next display.
+ */
+void CardInfoPictureWidget::updatePixmap()
+{
+    pixmapDirty = true;
+    update();
+}
+
+/**
+ * @brief Loads the appropriate pixmap based on the current card info.
+ *
+ * If `info` is valid, loads the card's image. Otherwise, loads a default card back image.
+ */
+void CardInfoPictureWidget::loadPixmap()
+{
+    if (info)
+        PictureLoader::getPixmap(resizedPixmap, info, size());
+    else
+        PictureLoader::getCardBackPixmap(resizedPixmap, size());
+
+    pixmapDirty = false;
+}
+
+/**
+ * @brief Custom paint event that draws the card image with rounded corners.
+ * @param event The paint event (unused).
+ *
+ * Checks if the pixmap needs to be reloaded. Then, calculates the size and position for centering the
+ * scaled pixmap within the widget, applies rounded corners, and draws the pixmap.
+ */
+void CardInfoPictureWidget::paintEvent(QPaintEvent *)
+{
+    if (width() == 0 || height() == 0)
+        return;
+
+    if (pixmapDirty)
+        loadPixmap();
+
+    QSize scaledSize = resizedPixmap.size().scaled(size(), Qt::KeepAspectRatio);
+    QPoint topLeft{(width() - scaledSize.width()) / 2, (height() - scaledSize.height()) / 2};
+    qreal radius = 0.05 * scaledSize.width();
+
+    QStylePainter painter(this);
+    QPainterPath shape;
+    shape.addRoundedRect(QRect(topLeft, scaledSize), radius, radius);
+    painter.setClipPath(shape);
+    painter.drawItemPixmap(QRect(topLeft, scaledSize), Qt::AlignCenter, resizedPixmap);
+}
+
+/**
+ * @brief Provides the recommended size for the widget based on the scale factor.
+ * @return The recommended widget size.
+ */
+QSize CardInfoPictureWidget::sizeHint() const
+{
+    return QSize(baseWidth * scaleFactor, baseHeight * scaleFactor);
+}
+
+/**
+ * @brief Starts the hover timer to show the enlarged pixmap on hover.
+ * @param event The enter event.
+ */
+void CardInfoPictureWidget::enterEvent(QEnterEvent *event)
+{
+    QWidget::enterEvent(event);
+    if (hoverToZoomEnabled) {
+        hoverTimer->start(500);
+    }
+
+    emit hoveredOnCard(info);
+}
+
+/**
+ * @brief Stops the hover timer and hides the enlarged pixmap when the mouse leaves.
+ * @param event The leave event.
+ */
+void CardInfoPictureWidget::leaveEvent(QEvent *event)
+{
+    QWidget::leaveEvent(event);
+    if (hoverToZoomEnabled) {
+        hoverTimer->stop();
+        enlargedPixmapWidget->hide();
+    }
+}
+
+/**
+ * @brief Moves the enlarged pixmap widget to follow the mouse cursor.
+ * @param event The mouse move event.
+ */
+void CardInfoPictureWidget::mouseMoveEvent(QMouseEvent *event)
+{
+    QWidget::mouseMoveEvent(event);
+    if (hoverToZoomEnabled && enlargedPixmapWidget->isVisible()) {
+        QPointF cursorPos = QCursor::pos();
+        enlargedPixmapWidget->move(QPoint(cursorPos.x() + 10, cursorPos.y() + 10));
```

Comment:
What's the +10 do?

Reply:
Offsets it slightly so it's a bit more underneath the cursor. It's slightly more visually pleasant.